### PR TITLE
Add folder argument to files:scan-app-data

### LINKED ISSUE #12327: files:scan-appdata should take folder argument
Sometimes you want to rescan your appdata, beacuse some previes got removed or you want to invalidate the js cache for whatever reason. Now the only option is to rescan everything. Which is kind of overkill if you only want to scan 1 tiny piece.

### PR DESCRIPTION
**Pull request by rullzer**
Fix #12327

If you need to rescan something but don't feel like rescanning your
whole preview stash.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]
:tada:

**Comment by kesselb** [comment]
-1 for your backlog ;)

File: apps/files/lib/Command/ScanAppData.php [hunk 1]
```diff
@@ -67,6 +68,8 @@ protected function configure() {
 		$this
 			->setName('files:scan-app-data')
 			->setDescription('rescan the AppData folder');
```
Comment (kesselb) [inline apps/files/lib/Command/ScanAppData.php:70]:
```suggestion
			->setDescription('rescan the AppData folder')
			->addArgument('folder', InputArgument::OPTIONAL, 'The appdata subfolder to scan', '');
```

**Review by ChristophWurst — APPROVED** [review]

**Comment by kesselb** [comment]
LGTM :+1: 

I have some small suggestion while you're at it ;) 

1) Newer versions of symfony/console requires that commands return a integer always: https://github.com/symfony/symfony/issues/33747

image

2) Don't show the statistics if folder does not exist and return with a non zero exit code.

**Comment by rullzer** [comment]
@kesselb mind to tackle them or file an issue? It is not really high on my priority list at the moment

**Comment by szaimen** [comment]
A question on this topic:

> If you need to rescan something but don't feel like rescanning your
> whole preview stash.

That sounds like based on this PR you are now able to exclude the preview folder while scanning the whole app-data folder. Is that the case or do I have to scan all other subfolders manually to be able to exclude the preview folder?

**Comment by rullzer** [comment]
The latter. There is no exclusion just selection of which folder to scan.

**Comment by szaimen** [comment]
Ok, thanks for the answer!

**Pull request merged**